### PR TITLE
Makefile: Fix compiling with deprecated code

### PR DESCRIPTION
**Pull request by masterleinad**
Addressing https://github.com/kokkos/kokkos/pull/7393/files#r1797275074.

### CODE REVIEW

File: Makefile.targets [hunk 1]
```diff
@@ -16,8 +16,6 @@ Kokkos_HostSpace.o: $(KOKKOS_CPP_DEPENDS) $(KOKKOS_PATH)/core/src/impl/Kokkos_Ho
 	$(CXX) $(KOKKOS_CPPFLAGS) $(KOKKOS_CXXFLAGS) $(CXXFLAGS) -c $(KOKKOS_PATH)/core/src/impl/Kokkos_HostSpace.cpp
 Kokkos_hwloc.o: $(KOKKOS_CPP_DEPENDS) $(KOKKOS_PATH)/core/src/impl/Kokkos_hwloc.cpp
 	$(CXX) $(KOKKOS_CPPFLAGS) $(KOKKOS_CXXFLAGS) $(CXXFLAGS) -c $(KOKKOS_PATH)/core/src/impl/Kokkos_hwloc.cpp
-Kokkos_TaskQueue.o: $(KOKKOS_CPP_DEPENDS) $(KOKKOS_PATH)/core/src/impl/Kokkos_TaskQueue.cpp
-	$(CXX) $(KOKKOS_CPPFLAGS) $(KOKKOS_CXXFLAGS) $(CXXFLAGS) -c $(KOKKOS_PATH)/core/src/impl/Kokkos_TaskQueue.cpp
```
Comment (dalg24) [inline Makefile.targets:20]: Holly smoke, deleted in d391bd9158cc212785565e391f51e991d2ec751c

File: Makefile.kokkos [hunk 2]
```diff
@@ -1624,6 +1624,12 @@ $(DESUL_CONFIG_HEADER):
 
 KOKKOS_CPP_DEPENDS := $(DESUL_CONFIG_HEADER) KokkosCore_config.h $(KOKKOS_HEADERS)
 
+# Tasking is deprecated
```
Comment (dalg24) [inline Makefile.kokkos:1627]:
```suggestion
# Tasking is deprecated.  Makefiles will be next...
```

**Review by dalg24 — APPROVED** [review]

**Comment by dalg24** [comment]
Retest this please

**Comment by dalg24** [comment]
Retest this please

**Review by crtrott — APPROVED** [review]

**Pull request merged**